lib/archethic_cache/lru_disk.ex

```diff
@@ -0,0 +1,69 @@
+defmodule ArchethicCache.LRUDisk do
+  @moduledoc """
+  Wraps the LRU genserver and adds hooks to write / read from disk.
+  The value is always a binary.
+  """
+  alias ArchethicCache.LRU
+
+  require Logger
+
+  @spec start_link(GenServer.name(), non_neg_integer(), binary()) ::
+          {:ok, binary()} | {:error, term()}
+  def start_link(name, max_bytes, cache_dir) do
+    cache_dir = Path.join(cache_dir, "#{name}")
+    :ok = reset_directory(cache_dir)
+
+    LRU.start_link(name, max_bytes,
+      put_fn: fn key, value ->
+        # retry to create the dir everytime in case someones delete it
+        File.mkdir_p!(cache_dir)
```

Comment:
I didn't get the `someones delete it`.
If the folder doesn't exist, the LRU would crash and restart from fresh state.

I'm afraid of the overhead trying to recreate a folder each time an entry is added